st2-ask/notification_rule.yaml

```diff
@@ -0,0 +1,21 @@
+---
+name: "notify_slack_execution_change"
+pack: "st2"
+description: "Watches for an execution status of "awaits" for a certain action ref
+enabled: true
+
+trigger:
+  type: core.st2.generic.actiontrigger
+
+criteria:
+  trigger.action_ref:
+      type: equals
+      pattern : st2.ask
+  trigger.status:
+      type: equals
+      pattern : awaits
+
+action:
+  ref: slack.post_message
+  parameters:
+    message: '{{trigger.execution_id}} is awaiting an approval action'
```

Comment:
The `core.st2.generic.actiontrigger` trigger gives a lot of information about the execution that caused the trigger, including params and result. I'm not sure if result will be populated when this trigger happens for executions in `awaits` or `pending` statuses, but params will be. Either could feasibly be added to `criteria`